internal/modifier/gated.go

```diff
@@ -78,5 +79,10 @@ func NewFeatureGatedModifier(logger logger.Interface, cfg *config.Config, image
 		discoverers = append(discoverers, d)
 	}
 
+	if !cfg.Features.AllowCUDACompatLibsFromContainer.IsEnabled() {
```

Comment:
Question -- are `features` in the config only applicable to libnvidia-container? If yes, how can we make this more clear to users? From the user's perspective, it may be a bit confusing that compat libs do end up being used even if `features.allow-cuda-compat-libs-from-container=false` is set in the config.

Reply:
`features` are not strictly limited to `libnvidia-container`. Note that `libnvidia-container` does not process any of these features and they are ONLY checked from the `nvidia-container-runtime` or `nvidia-container-runtime-hook`. I think this is a side-effect of us not being specific enough when naming the feature flag in this case. Since features are not listed in the config by default, and only listed if toggled explicitly, I think we should be ok with this slight confusion.